Android/app/src/go/intra/split/direct_split.go

```diff
@@ -54,13 +54,20 @@ func (s *splitter) Write(b []byte) (int, error) {
 
 	// Setting `used` to true ensures that this code only runs once per socket.
 	s.used = true
-	b1, b2 := splitHello(b)
-	n1, err := conn.Write(b1)
-	if err != nil {
-		return n1, err
+	pkts, _ := splitHello(b)
```

Comment:
We will need to pass a writer to the splitHello to avoid allocations, since we are adding a TLS record header.

Reply:
done